lib/runner.go

```diff
@@ -90,4 +90,11 @@ type Runner interface {
 	// values and write it back to the runner.
 	GetOptions() Options
 	SetOptions(opts Options) error
+
+	// GetExports returns the names of exported functions in the script
+	// (excluding setup() and teardown()) that can be used for execution.
```

Comment:
I would prefer if we do it the other way around, at least for now. 
Have a function `isThisExportedFunction` which checks if "myCustomeDefault" is an approriately typed exported symbol and return a boolean. 

Maybe the name should be `IsExecable`  as it checks more stuff ... 
We can also have Exec defaulting to `default` which will remove the need to have it being checked in two places, but will mean that exec will need to be null.String in the executor configs

Reply:
actually it already is null.String so if we can find a way to set it's default ... we are ready. 

Also the above will mean that there will be no need to get all the exported symbols previously to execution, which IMO is better. It will mean we lose the "no exported functions in script" error, but I don't think that is much better then telling them there is no `default` function being exported ... if anything it seems to be worse UX IMO

Reply:
> Hmm why should we restrict that?

My thinking was for this to be used only for functions that a VU can use for execution, since `setup()` and `teardown()` are handled separately (with `js.Runner.runPart()`, which I'd prefer to get rid of in favor of a unified `Run*` function), so using it for other exported objects doesn't make sense currently. I suppose we could refactor this now if you have a use case in mind we should support in the short term.

> Moreover, it might not be a bad idea to specify the type of the exported thing

The reason the values are `struct{}` right now is to avoid a circular import (`lib` <-> `js`), and since for validation we don't really need to have access to the exported objects. A bit janky, but I don't see a benefit to changing the `lib.Runner` interface because of this.

> Have a function `isThisExportedFunction` which checks if "myCustomeDefault" is an approriately typed exported symbol and return a boolean.

Yeah, this could work as well, but when would we check this? I'd be against checking this after VUs are initialized, since we should fail earlier than that. Maybe a good candidate for `BaseExecutor.Init()`? I'll consider this and probably implement it before tomorrow's meeting, so we can discuss it then.

I pushed some changes now in 4d474454 to do the `Exec` fallback in `BaseConfig.GetExec()`, since I couldn't find a way to set a default for `null.String` values. It's not great since it hides the original user setting, ~but at least it's done in a single place.~ No, we still need it in `Activate()` too, mainly for the tests that don't use a `Config`, otherwise there would be a bunch of `Exec: consts.DefaultFn` in all tests. :-/

Reply:
in the exact same place https://github.com/loadimpact/k6/pull/1428/files#diff-31f9f805fb3e45c114bdaa74f9447759R297 , just you will provide a function `isExecFnAble(str name) bool` instead of an object that needs generation.  (name up for debate ;) )

The function will just be `Runner.IsExecFnAble` bound to the runner in question (you can just give it as `r.IsExecFnAble` as long as `r` is of implementing Runner ;) ). 
This saves us the whole generation of exports - the `js.Runner` can decide in its implementation what is a runnable symbol (so for example teardown won't be ;) ). 

Also this will mean that we don't need to have arguably `js` constants outside of the `js` package. I am not certain how much of a moot point this is as we are very unlikely to add other runners anytime soon, and they arguably will have some kind of setup/teardown/options variant as well ?

Reply:
Good idea, see 9c1d408f.

I named it `IsExecutable` for simplicity and ease of pronunciation :). Even though it's on `Runner`, the fact it takes a `name string` should imply it's checking the script for executable functions and not the runner itself. This is better than passing the exports object around, but still not ideal that this needs to be on the `Runner`, so maybe something to refactor in the future.

> Also this will mean that we don't need to have arguably js constants outside of the js package.

This change doesn't really address this, and while I agree that `lib/consts/js.go` is a messy place for these constants, I much prefer that over having to repeat them everywhere. But let me know if you disagree, and I'll revert f7bc0fa7.

@na-- 9c1d408f also now grabs `setup`/`teardown`. You're right that that might be useful for the scenario of using a `iterations: 1, exec: "setup"` executor for setup, for example. Though I still don't see the scenario of having the `options` object being useful. :)